qui/updater/intro_page.py

```diff
@@ -0,0 +1,349 @@
+# Copyright (C) 2023  Piotr Bartman <prbartman@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+import subprocess
+from datetime import datetime, timedelta
+from typing import Union, Optional
+
+from qubes_config.widgets.gtk_utils import load_icon
+from qubesadmin import exc
+
+from qui.updater.utils import disable_checkboxes, HeaderCheckbox, \
+    pass_through_event_window, QubeLabel, Theme, \
+    QubeName, label_color_theme, UpdateStatus, RowWrapper, \
+    ListWrapper, on_head_checkbox_toggled
+
+
+class IntroPage:
+    """
+    First content page of updater.
+
+    Show the list of updatable vms with an update info.
+    """
+
+    def __init__(self, builder, theme, next_button):
+        self.builder = builder
+        self.theme = theme
+        self.next_button = next_button
+        self.disable_checkboxes = False
+        self.active = True
+
+        self.page = self.builder.get_object("list_page")
+        self.stack = self.builder.get_object("main_stack")
+        self.vm_list = self.builder.get_object("vm_list")
+        self.list_store: Optional[ListWrapper] = None
+
+        checkbox_column = self.builder.get_object("checkbox_column")
+        checkbox_column.connect("clicked", self.on_header_toggled)
+        header_button = checkbox_column.get_button()
+
+        header_button.connect('realize', pass_through_event_window)
+
+        self.checkbox_column_button = self.builder.get_object("checkbox_header")
+        self.checkbox_column_button.set_inconsistent(True)
+        self.checkbox_column_button.connect("toggled", self.on_header_toggled)
+        self.head_checkbox = HeaderCheckbox(
+            self.checkbox_column_button,
+            allowed=["YES", "MAYBE", "NO"],
+            callback_all=lambda *_args: self.next_button.set_sensitive(True),
+            callback_some=lambda *_args: self.next_button.set_sensitive(True),
+            callback_none=lambda *_args: self.next_button.set_sensitive(False),
+        )
+
+        self.vm_list.connect("row-activated", self.on_checkbox_toggled)
+
+        self.info_how_it_works = self.builder.get_object("info_how_it_works")
+        self.info_how_it_works.set_label(
+            self.info_how_it_works.get_label().format(
+                MAYBE='<span foreground="Orange"><b>MAYBE</b></span>'))
```

Comment:
could this be done with styles? It's just a label. I'd rather put the color in stylesheets if possible, to be able to easily adjust it if dark/light mode needs it.